### ISSUE TITLE
Tailor as proxy during template fetching

### ISSUE THREAD

**Issue by tomaszglinski** [zalando/tailor#252]
>  If your front-end team is making the monolith-to-microservices transition, you might find Tailor and its available siblings beneficial

Hi, so we are doing exactly that :) But we found one big problem with Tailor. Since we want to do the transition in a step-by-step manner, we need in fact Tailor to work as a proxy, where the proxying result HTML is the potential template  (potential means: not necesarilly there are any fragments within).
So we tried to put it in front of the monolith, and unfortunately it didn't go well. The main problem is handling more complex requests, like POSTs, especially if files are to be transferred. Tailor simply does not look to have such functionality, right? Do we need to write our own fethTemplate function, that would handle all the HTTP traffic correctly?

Also the question is, how do you handle that kind of pages on Zalando? Or you just limit Tailor usage to GET requests? I found that Corporate section of you sites are without Tailor, so you don't use it in general...

**Comment by clitcancer** [comment]
Don't forget that Tailor is only a layout service.
Check out [skipper](https://github.com/zalando/skipper), it's zalando's router, integrates very nicely with Tailor. Perfect for the mono-to-micro transition. Setup  routes that should use Tailor and other that should use your old app. It's a reverse proxy too, so exactly what you're looking for :)

**Comment by tomaszglinski** [comment]
@clitcancer I red about skipper, I didn't really find the solution to the problem. Skipper sound to be simply a routing service, I don't see how does it helps Tailor to handle POSTs.
Am I correct, that in Zalando architecture non-"GET *.html" requests are treated as API calls and go directly to services (no Tailor on the way), and do not render pages in return (because they could not contain fragments, as there is no Tailor on the way)?

**Comment by clitcancer** [comment]
To clarify, you want Tailor to handle POSTs? because Tailor should be used for rendering your fragments (microfrontends) into a complete page (with GETs). 
If your goal is to have a POST service on the same server as tailor you could always have (Express example)
```js
app.get('/*', tailor.requestHandler) // all GETs go to tailor
app.post('/yourPath', (req, res) => {/* handle */}) // POSTs handled by your callbacks (not related to Tailor)
```

**Comment by tomaszglinski** [comment]
Hi @clitcancer , let me rephrase :) 
How do you handle the case, where backend server have a page with form within, submission of that form issues a POST request. In the response we would like to see a Tailored page (with header fragment)?

**Comment by clitcancer** [comment]
(Assuming you know what are you doing when expecting a html page after a POST request)
Your POST request handler could always just make a GET to Tailor and send it back as the POST response.

**Comment by vigneshshanmugam** [comment]
closing as question is answered.
